platform/src/main/java/org/stellar/anchor/platform/configurator/DataConfigAdapter.java

```diff
@@ -0,0 +1,182 @@
+ * spring.jpa.generate-ddl:                            Whether to initialize the schema on startup. : false
+ * spring.jpa.hibernate.ddl-auto:                      DDL mode. This is actually a shortcut for the "hibernate.hbm2ddl.auto" property. Defaults to "create-drop" when using an embedded database and no schema manager was detected. Otherwise, defaults to "none".
+ * spring.jpa.hibernate.naming.implicit-strategy:      Fully qualified name of the implicit naming strategy.
+ * spring.jpa.hibernate.naming.physical-strategy:      Fully qualified name of the physical naming strategy.
+ * spring.jpa.hibernate.use-new-id-generator-mappings: Whether to use Hibernate's newer IdentifierGenerator for AUTO, TABLE and SEQUENCE. This is actually a shortcut for the "hibernate.id.new_generator_mappings" property. When not specified will default to "true".
+ * spring.jpa.mapping-resources:                       Mapping resources (equivalent to "mapping-file" entries in persistence.xml).
+ * spring.jpa.open-in-view:                            Register OpenEntityManagerInViewInterceptor. Binds a JPA EntityManager to the thread for the entire processing of the request. : true
+ * spring.jpa.show-sql:                                Show SQL : false
+ * </pre>
+ */
+public class DataConfigAdapter extends SpringConfigAdapter {
+  List<String> allFields =
+      Arrays.asList(
+          "spring.datasource.driver-class-name",
+          "spring.datasource.name",
+          "spring.datasource.type",
+          "spring.datasource.url",
+          "spring.datasource.username",
+          "spring.datasource.password",
+          "spring.datasource.generate-unique-name",
+          "spring.datasource.hikari.connection-timeout ",
+          "spring.datasource.hikari.minimum-idle",
+          "spring.datasource.hikari.maximum-pool-size",
+          "spring.datasource.hikari.idle-timeout",
+          "spring.datasource.hikari.max-lifetime",
+          "spring.datasource.hikari.auto-commit ",
+          "spring.jpa.database",
+          "spring.jpa.database-platform",
+          "spring.jpa.defer-datasource-initialization",
+          "spring.jpa.generate-ddl",
+          "spring.jpa.hibernate.use-new-id-generator-mappings",
+          "spring.jpa.open-in-view",
+          "spring.jpa.show-sql",
+          "spring.flyway.enabled");
+
+  @Override
+  void sendToSpring(ConfigMap config) throws InvalidConfigException {
+    // Set our default value to start with
+    setSpringDataDefaults();
+
+    //
+    // The following code block purposely kept long for easier understanding of how the Spring
+    // database configurations are set.
+    //
+    String type = config.getString("data.type").toLowerCase();
+    switch (type) {
+      case "h2":
+        set("spring.datasource.driver-class-name", "org.h2.Driver");
```

Comment:
This is vague to me... I know there are code repetitions at this moment. It is still vague to me how much value function extraction would bring at this moment. 

If we encounter another repetition for `flyway`, I will extract. 

The rest does not worth extraction at this moment. But it may worth extraction in the future.